yolo.py

```diff
@@ -153,33 +155,58 @@ def set_api_key():
     temperature=config["temperature"],
     max_tokens=config["max_tokens"],
   )
+ 
+  return response.choices[0].message.content.strip()
 
-#print (response)
-
-res_command = response.choices[0].message.content.strip()
 
 #Enable color output on Windows using colorama
 init() 
 
-prefixes = ("sorry", "i'm sorry", "the question is not clear", "i'm", "i am")
-if res_command.lower().startswith(prefixes):
-  print(colored("There was an issue: "+res_command, 'red'))
-  sys.exit(-1)
-
-# odd corner case, sometimes ChatCompletion returns markdown
-if res_command.count("```",2):
-  print(colored("The proposed command contains markdown, so I did not execute the response directly: \n", 'red')+res_command)
-  sys.exit(-1)
+def check_for_issue(response):
+  prefixes = ("sorry", "i'm sorry", "the question is not clear", "i'm", "i am")
+  if response.lower().startswith(prefixes):
+    print(colored("There was an issue: "+response, 'red'))
+    sys.exit(-1)
 
-print("Command: " + colored(res_command, 'blue'))
-if config["safety"] != "off" or ask_flag == True:
-  print("Execute the command? [Y/n] ==> ", end = '')
-  yolo = input()
-  print()
+def check_for_markdown(response):
+  # odd corner case, sometimes ChatCompletion returns markdown
+  if response.count("```",2):
+    print(colored("The proposed command contains markdown, so I did not execute the response directly: \n", 'red')+response)
+    sys.exit(-1)
 
-if yolo.upper() == "Y" or yolo == "":
-  if shell == "powershell.exe":
-    subprocess.run([shell, "/c", res_command], shell=False)  
-  else: 
-    # Unix: /bin/bash /bin/zsh: uses -c both Ubuntu and macOS should work, others might not
-    subprocess.run([shell, "-c", res_command], shell=False)
\ No newline at end of file
+def prompt_user_input(response):
+  print("Command: " + colored(response, 'blue'))
+  if config["safety"] != "off" or ask_flag == True:
+    print("Execute command? [Y]es [n]o [m]odify [c]opy to clipboard ==> ", end = '')
+    user_input = input()
+    return user_input 
+
+def evaluate_input(user_input, command):
+  if user_input.upper() == "Y" or user_input == "":
+    if shell == "powershell.exe":
+      subprocess.run([shell, "/c", command], shell=False)  
+    else: 
+      # Unix: /bin/bash /bin/zsh: uses -c both Ubuntu and macOS should work, others might not
+      subprocess.run([shell, "-c", command], shell=False)
+  
+  if user_input.upper() == "M":
+    print("Modify prompt: ", end = '')
+    modded_query = input()
+    modded_response = call_open_ai(modded_query)
+    print(modded_response)
+    check_for_issue(modded_response)
+    check_for_markdown(modded_response)
+    modded_user_input = prompt_user_input(modded_response)
+    print()
+    evaluate_input(modded_user_input, modded_response)
+  
+  if user_input.upper() == "C":
+    pyperclip.copy(command) 
```

Comment:
I'm mostly working with VS Code Remote Ceveloper and SSH to machines.  In that case I noticed that there is an error with pyperclip. I think it needs a GUI to work/have a clipboard to copy to.

The exact error I'm getting is: 
```
 Pyperclip could not find a copy/paste mechanism for your system.
    For more information, please visit https://pyperclip.readthedocs.io/en/latest/index.html#not-implemented-error 
```

Not sure if there is an easy way we could only show the (c)opy option when it conditions are set to work. 

Otherwise something like this would suffice probably also:
```
if user_input.upper() == "C":
    try:
      pyperclip.copy(command) 
      print("Copied command to clipboard.")
    except:
      print(colored("Error: Clipboard copy failed. You are probably in an SSH session and there is n>
```

Let me know your thoughts?

Reply:
pushed some updates to check for a display and adjust the input prompt accordingly. Think it should work for posix machines but I'm not really sure how that would work on a windows server. 
I think the try/except block is probably a good idea to handle any potential errors more gracefully.

Reply:
Good question about Windows without a GUI or with PowerShell Remoting (e.g. `PSEnter-Session`). I'll do some testing in this regards down the road, but I think it's a less common scenario.

Again, really appreciate the feature additions.